pisa/core/translation.py

```diff
@@ -361,18 +361,31 @@ def find_index(x, bin_edges):
     direct transformations instead of search
     """
     # TODO: support lin and log binnings with
-
-    first = 0
-    last = len(bin_edges) - 1
-    while first <= last:
-        i = int((first + last)/2)
-        if x >= bin_edges[i]:
-            if (x < bin_edges[i+1]) or (x <= bin_edges[-1] and i == len(bin_edges) - 1):
-                break
+    
+    #
+    # First check: extreme bin
+    #
+    if x<bin_edges[0]:
+        return -1
+    elif x>bin_edges[-1]:
+        return len(bin_edges)-1
```

Comment:
shouldn't this be `return len(bin_edges)`?

Reply:
the overflow bin has a value of nbin, which has one unit less than the length of the bin edges

Reply:
if there are N bins, we have N+1 bin edges. The last bin has index N-1, so the overflow should be N = `len(bin_edges) - 1`, if i did my math correctly?

Reply:
ah ok, we were typing answers at the same time ;)

Reply:
Very well, my mistake. I'm adding some variables for clarity in my PR.